pkg/controller/events/workflow_execution.go

```diff
@@ -0,0 +1,79 @@
+package events
+
+import (
+	"context"
+	"strings"
+
+	"github.com/flyteorg/flyteidl/clients/go/events"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/event"
+	"github.com/flyteorg/flytepropeller/pkg/controller/config"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/flyteorg/flytestdlib/storage"
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+//go:generate mockery -all -output=mocks -case=underscore
+
+// Recorder for Workflow events
+type WorkflowEventRecorder interface {
+	RecordWorkflowEvent(ctx context.Context, event *event.WorkflowExecutionEvent, outputPolicy config.RawOutputPolicy) error
+}
+
+type workflowEventRecorder struct {
+	eventRecorder events.WorkflowEventRecorder
+	store         *storage.DataStore
+}
+
+// In certain cases, a successful workflow execution event can be configured to include raw output data inline. However,
+// for large outputs these events may exceed the event recipient's message size limit, so we fallback to passing
+// the offloaded output URI instead.
+func (r *workflowEventRecorder) handleFailure(ctx context.Context, ev *event.WorkflowExecutionEvent, err error, rawOutputPolicy config.RawOutputPolicy) error {
+	// Only attempt to retry sending an event in the case we tried to send raw output data inline
+	if rawOutputPolicy != config.RawOutputPolicyInline || len(ev.GetOutputUri()) > 0 {
```

Comment:
how will this work? should we just fail the workflow in this case? I dont think this is right.

Lets say some deployment creates a large output, but we just cannot send it, then offloading will not help

Reply:
i think it will be better to fail the workflow with right error message?

Reply:
this was on a suggestion from Haytham: https://github.com/flyteorg/flytepropeller/pull/304#issuecomment-901315689
maybe we can make the retry configurable based on the propeller set-up via a config option

Reply:
updated to make configurable